src/components/content/index.js

```diff
@@ -0,0 +1,33 @@
+import React, { useState, useEffect } from 'react';
+
+import { getTopics, voteTopic } from '../../services/topics';
+import EmptyList from '../empty-list';
+import TopicList from '../topic-list';
+
+const Content = () => {
+  const [topics, setTopics] = useState();
+
+  const fetchTopics = async () => {
+    const response = await getTopics();
+    setTopics(response);
+  };
+
+  const onVote = async (id, amount) => {
+    await voteTopic({ id, amount });
+    fetchTopics();
+  };
+
+  useEffect(() => {
+    fetchTopics();
+  }, []);
+
+  return topics ? (
+    topics.length ? (
+      <TopicList {...{ topics, onVote }} />
```

Comment:
Do you see any issue passing onVote like this?

Reply:
You're right, this will cause some unnecessary re-renders in the children components. I think I could add `useCallback` here, or introduce Redux in order to update the list using one-way data flow.